### PR TITLE
feat(core): consents for resource

### PR DESCRIPTION
**Pull request by sarkapalkovicova**
Added methods to obtain consents for members assigned to the chosen resource.

### CODE REVIEW

File: perun-core/src/test/java/cz/metacentrum/perun/core/entry/ConsentsManagerEntryIntegrationTest.java [hunk 1]
```diff
@@ -197,6 +197,21 @@ public void getConsentsForUser() throws Exception {
 		assertEquals(0, consentsManagerEntry.getConsentsForUser(sess, user.getId(), ConsentStatus.GRANTED).size());
 	}
 
+	@Test
+	public void getConsentsForConsentHubByResource() throws Exception {
```
Comment (xflord) [inline perun-core/src/test/java/cz/metacentrum/perun/core/entry/ConsentsManagerEntryIntegrationTest.java:201]: I would rather have this test create a resource under a facility, assign some members to it, create consents for them and then assert that the method only returns the consents of these assigned members.

File: perun-core/src/main/java/cz/metacentrum/perun/core/api/ConsentsManager.java [hunk 2]
```diff
@@ -49,6 +50,19 @@ public interface ConsentsManager {
 	 */
 	List<Consent> getConsentsForConsentHub(PerunSession sess, int id) throws PrivilegeException, ConsentHubNotExistsException;
 
+	/**
+	 * Get all consents for ConsentHub corresponding to the facility of the chosen resource.
```
Comment (xflord) [inline perun-core/src/main/java/cz/metacentrum/perun/core/api/ConsentsManager.java:54]: also edit the docs. (we want to return only the consents of the members of that resource, not all consents for the facility)

**Review by xflord — APPROVED** [review]
well done

File: perun-base/src/main/resources/perun-roles.yml [hunk 3]
```diff
@@ -7934,6 +7934,16 @@ perun_policies:
     include_policies:
       - default_policy
 
+  getConsentHubByResource_Resource_policy:
+    policy_roles:
+      - PERUNOBSERVER:
+      - FACILITYADMIN: Facility
+      - FACILITYOBSERVER: Facility
+      - VOADMIN: Vo
+      - VOOBSERVER: Vo
```
Comment (Kuliak) [inline perun-base/src/main/resources/perun-roles.yml:7943]: Shouldn't we allow also RESOURCEADMIN and RESOURCEOBSERVER to use this method?

**Review by Kuliak — COMMENTED** [review]
Good job! I have only one suggestion to check.

**Review by mattjoke — APPROVED** [review]
:+1:

**Review by Kuliak — APPROVED** [review]

**Pull request merged**